src/main/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,84 @@
+package calculator;
+
+import static java.util.Arrays.*;
+import static java.util.Objects.isNull;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+    private static final String DEFAULT_DELIMITER_REGEX = ",|:";
+    private static final String CUSTOM_DELIMITER_REGEX = "//(.)\n(.*)";
+
+    public int splitAndSum(String input) {
+        if (isNotExist(input)) {
+            return 0;
+        }
+
+        // 숫자가 하나인 경우에도 커스텀 관련 인풋은 인입될 수 있다는 가정하에
+        // 계산 대상 숫자목록 추출작업후 1개 여부 관련 요구사항을 처리했습니다
+        String[] targetNumbers = getNumbers(input);
+        if (hasJustOneNum(targetNumbers)) {
+            return processOneNumber(targetNumbers[0]);
+        }
+
+        return stream(targetNumbers)
+                .mapToInt(this::processOneNumber)
+                .sum();
+    }
+
+    private int processOneNumber(String targetNumber) {
+        if (isNotDigit(targetNumber)) {
+            throw new RuntimeException("입력값은 숫자여야 합니다");
+        }
+
+        int number = toInt(targetNumber);
+
+        if (isNotPositive(number)) {
+            throw new RuntimeException("입력값은 양수여야 합니다");
+        }
+
+        return number;
+    }
+
+    private String[] getNumbers(String input) {
+        Matcher matcherResult = Pattern.compile(CUSTOM_DELIMITER_REGEX).matcher(input);
+        if (matcherResult.find()) {
+            String delimiter = getCustomDelimiter(matcherResult);
+            String numbers = polishInput(matcherResult);
```

Comment:
이 메서드의 다른 코드는 추상화 레벨이 0이고, 이 두 부분은 추상화 레벨이 1이다.
굳이 메서드로 분리하기보다 아래와 같이 구현하는 것은 어떻게 생각하나?
어느 부분을 메서드로 분리하는 것이 의미있을지 한번 고민해 보면 어떨까?

```suggestion
            String delimiter = matcherResult.group(1);
            String numbers = matcherResult.group(2);
```

Reply:
피드백 감사합니다 튜터님.
위 상황에선 말씀하신대로 변수명이 해당 동작에 대한 설명을 충분히 할수 있을것 같아 추상화레벨은 동일하게 가져가는게 좋다는 생각이 듭니다

추가적으로 이런 메서드 분리/클래스 분리에 대한 명확한 기준은 실무에서도 항상 고민되었던 부분인데요.
음.. 제 생각엔 기본적으로는 추상화레벨을 맞추되
1. 도메인적으로 중요한 개념,
2. 2~3 라인이지만 외부 라이브러리를 사용할때
3. 2~3 라인이지만 생소한 기능(ex 복잡한 스트림)or 중요한 핵심 기능일때
위의 케이스에선 추상화 레벨을 올려주는것도 괜찮다는 생각이 드네요

항상 이럴때 마다 저만의 명확한 기준들이 없어서 그때그때 다른 기준을 적용하게 되는것 같아요
어떤날은 A 기준으로 추상화.. 어떤 날은 B 기준으로 추상화..  이렇다보니 제 클린코딩 실력도 늘지 않는것 같다는 생각을 하고 있습니다

뭔가 주절주절 말이 길었던거 같은데..ㅎ..
결론적으로 여쭙고 싶은건 튜터님은 메서드 분리에 어떤 기준을 가지고 계시는지가 궁금합니다